test/test_util.py

```diff
@@ -123,7 +123,9 @@ def test_invalid_host(self):
         'http://foo:bar@localhost/': Url('http', auth='foo:bar', host='localhost', path='/'),
         'http://foo@localhost/': Url('http', auth='foo', host='localhost', path='/'),
         'http://foo:bar@baz@localhost/': Url('http', auth='foo:bar@baz', host='localhost', path='/'),
-        'http://@': Url('http', host=None, auth='')
+        'http://@': Url('http', host=None, auth=''),
+        'http://foo:bar#@localhost': Url('http', auth='foo:bar#', host='localhost'),
+        'http://foo:bar/baz@localhost': Url('http', auth='foo:bar/baz', host='localhost')
```

Comment:
Please add a test for the following:

```
http://localhost/@username
http://localhost:5000/?foo=@bar
http://localhost/#biz@baz
```

None of those should have auth parameters (in case that wasn't clear).